packages/manager/apps/web/client/app/hosting/multisite/hosting-multisite.service.js

```diff
@@ -104,44 +194,65 @@ angular.module('services').service(
       runtimeId,
       serviceName,
     ) {
-      return this.$http
-        .put(`${this.aapiHostingPath}/${serviceName}/domains`, {
-          baseDomain,
-          domainName,
-          home,
-          wwwNeeded,
-          ipv6Needed,
-          autoconfigure,
-          cdn: cdn.toLowerCase(),
-          countryIp,
-          firewallNeeded: firewall.toLowerCase(),
+      const completeDomain = domainName
+        ? `${domainName}.${baseDomain}`
+        : baseDomain;
+
+      const ipLocation = countryIp?.country ? countryIp.country : null;
+
+      const addDomain = this.RequestAttachedDomainCreate(
+        serviceName,
+        completeDomain,
+        cdn.toLowerCase(),
+        firewall.toLowerCase(),
+        ownLog,
+        home,
+        runtimeId,
+        ssl,
+        !autoconfigure,
+        ipLocation,
+      );
+
+      const promises = [addDomain];
+
+      if (wwwNeeded) {
+        const addDomainwww = this.RequestAttachedDomainCreate(
+          serviceName,
+          `www.${completeDomain}`,
+          cdn.toLowerCase(),
+          firewall.toLowerCase(),
           ownLog,
-          ssl,
+          home,
           runtimeId,
-        })
-        .then((response) => {
-          this.Hosting.resetDomains();
-          this.getTaskIds({ fn: 'attachedDomain/create' }, serviceName).then(
-            (taskIds) => {
-              this.pollRequest({
-                serviceName,
-                taskIds,
-                namespace: 'modifyDomain',
-              });
-            },
-          );
-          return response.data;
-        });
+          ssl,
+          !autoconfigure,
+          ipLocation,
+        );
+
+        promises.push(addDomainwww);
+      }
+
+      return Promise.all(promises).then(() => {
```

Comment:
**nitpick:** Instead of using the native JS Promise, better to use `$q.all` as there change detection issues when native JS Promises are used with AngularJS.